MRETestBed/Assets/TestBed Assets/Scripts/UserInfo.cs

```diff
@@ -28,8 +28,32 @@ public UserInfo(Guid id)
     public event MWEventHandler BeforeAvatarDestroyed;
     public event MWEventHandler AfterAvatarCreated;
 
+    private static Transform FindChildRecursive(Transform parent, string name)
+    {
+        Transform transform = parent.Find(name);
+        if (transform != null)
+        {
+            return transform;
+        }
+        for (int i = 0; i < parent.childCount; ++i)
+        {
```

Comment:
This can be a ranged for loop by doing `foreach (Transform t in parent.GetComponentsInChildren<Transform>())` which is cleaner.  Ranged for is always preferred over indexed for loop to range safety.

Reply:
For that matter, this whole method could be a one-liner: `parent.GetComponentsInChildren<Transform>().FirstOrDefault(t => t.name == name)`

---
In reply to: [261803994](https://github.com/Microsoft/mixed-reality-extension-unity/pull/42#discussion_r261803994) [](ancestors = 261803994)

Reply:
Yeah.  That too.